src/librustc/ty/print/pretty.rs

```diff
@@ -930,13 +930,34 @@ pub trait PrettyPrinter<'tcx>:
         let u8 = self.tcx().types.u8;
 
         match (scalar, &ty.kind) {
+            // Single element arrays just print their element (they are `#[transparent]`).
```

Comment:
Well, we do wrap it in `[...]` still... the comment reads like we would not do that.^^

I am not sure why single-element is treated specially but I guess this is pre-existing.

Reply:
Because they are easy to print as they have the same representation as the element itself. Printing multi-element arrays requires processing `Allocation`s

Reply:
comment updated